projects/ngx-d3timeline/src/lib/core/resize-observable.ts

```diff
@@ -0,0 +1,25 @@
+import { fromEventPattern } from 'rxjs';
+import { NodeEventHandler } from 'rxjs/internal/observable/fromEvent';
+
+declare var ResizeObserver: any; // typings not yet available in Typescript
+declare type ResizeObserver = any;
+
+export function createResizeObservable(element: Element) {
```

Comment:
I'm thinking the other two functions are private helper functions that can go inside this one maybe something like:

```typescript
export function createResizeObservable(element: Element) {
  function addResizeHandler(handler: NodeEventHandler) {
    const resizeObserver = new ResizeObserver(entries => {
      const entry = entries[0];
      handler([entry.contentRect.width, entry.contentRect.height]);
    });
    resizeObserver.observe(element);
    return resizeObserver;
  }

  function removeResizeHandler(token: ResizeObserver) {
    token.disconnect();
  }

  return fromEventPattern<[number, number]>(
    addResizeHandler,
    (_handler, token) => removeResizeHandler(token)
  );
}
```

Reply:
While valid it didn't cross my mind to use nesting here, not something I've done much if ever. 

Currently the 'private' comes out of them not being exported. IMO its easier to read without the nesting, there are just 3 discrete functions. 

It also equally could be a class with a constructor and private methods, note that ResizeObserver itself is a class.

In any case I can also see the relevance of nested functions, so if you want to go that way then I can change it.

Reply:
> Currently the 'private' comes out of them not being exported. IMO its easier to read without the nesting, there are just 3 discrete functions.

They are private because they only make sense in the context of `createResizeObservable`, i.e. they are just helper functions.

> In any case I can also see the relevance of nested functions, so if you want to go that way then I can change it.

I reckon better to change it

Reply:
done